ion-schema/src/system.rs

```diff
@@ -437,12 +439,12 @@ static DERIVED_ISL_TYPES: [&str; 10] = [
     "type::{ name: $lob, one_of: [ $blob, $clob ] }",
     "type::{ name: $number, one_of: [ $decimal, $float, $int ] }",
     "type::{ name: $text, one_of: [ $string, $symbol ] }",
-    "type::{ name: $any, one_of: [ $blob, $bool, $clob, $decimal,
-                                    $float, $int, $string, $symbol, $timestamp,
-                                    $list, $sexp, $struct, $null ] }",
     // this is just a place holder for document type,
     // IonSchemaElement::Document(_) type is used to verify the correctness on the validation side
     "type::{ name: document }",
+    "type::{ name: $any, one_of: [ $blob, $bool, $clob, $decimal,
+                                    $float, $int, $string, $symbol, $timestamp,
+                                    $list, $sexp, $struct, $null, document ] }",
```

Comment:
Corrects the definition for `$any` type which should also include `document`  type.